python_modules/libraries/dagster-aws/dagster_aws/athena/resources.py

```diff
@@ -196,47 +204,89 @@ def _fake_results(self, execution_id, expected_results):
+    polling_interval: int = Field(
+        default=5,
+        description=(
+            "Time in seconds between checks to see if a query execution is finished. 5 seconds"
+            " by default. Must be non-negative."
         ),
-        "max_polls": Field(
-            int,
-            description=(
-                "Number of times to poll before timing out. 120 attempts by default. When coupled"
-                " with the default polling_interval, queries will timeout after 10 minutes (120 * 5"
-                " seconds). Must be greater than 0."
-            ),
-            is_required=False,
-            default_value=120,
+    )
+    max_polls: int = Field(
+        default=120,
+        description=(
+            "Number of times to poll before timing out. 120 attempts by default. When coupled"
+            " with the default polling_interval, queries will timeout after 10 minutes (120 * 5"
+            " seconds). Must be greater than 0."
         ),
-        "aws_access_key_id": Field(StringSource, is_required=False),
-        "aws_secret_access_key": Field(StringSource, is_required=False),
-    }
+    )
+    aws_access_key_id: Optional[str] = Field(
+        default=None, description="AWS access key ID for authentication purposes."
+    )
+    aws_secret_access_key: Optional[str] = Field(
+        default=None, description="AWS secret access key for authentication purposes."
+    )
+
+
+class AthenaClientResource(ResourceWithAthenaConfig):
+    """This resource enables connecting to AWS Athena and issuing queries against it.
+
+    Example:
+        .. code-block:: python
+
+                from dagster import build_op_context, op
+                from dagster_aws.athena import AthenaClientResource
+
+                @op(required_resource_keys={"athena"})
+                def example_athena_op(context):
+                    return context.resources.athena.execute_query("SELECT 1", fetch_results=True)
+
+                context = build_op_context(resources={"athena": AthenaClientResource})
+                assert example_athena_op(context) == [("1",)]
+
+    """
+
+    def provide_object_for_execution(self, context: InitResourceContext) -> AthenaClient:
+        client = boto3.client(
+            "athena",
+            aws_access_key_id=self.aws_access_key_id,
+            aws_secret_access_key=self.aws_secret_access_key,
+        )
+        return AthenaClient(
+            client=client,
+            workgroup=self.workgroup,
+            polling_interval=self.polling_interval,
+            max_polls=self.max_polls,
+        )
+
+
+class FakeAthenaClientResource(ResourceWithAthenaConfig):
+    def provide_object_for_execution(self, context: InitResourceContext) -> FakeAthenaClient:
+        return FakeAthenaClient(
+            client=boto3.client("athena", region_name="us-east-1"),
+            workgroup=self.workgroup,
+            polling_interval=self.polling_interval,
+            max_polls=self.max_polls,
+        )
```

Comment:
This predates you, but I don't see how this is a fake. This is an Athena client hardcoded to talk to us-east-1

Reply:
Changed name to `TestHardcodedAthenaClientResource` & moved to test files for now. Not sure why this lived in the public package.

Reply:
It was Fake but only if you tuned on `moto` like we did in the tests, meaning it's not very useful out of the box for users.